### PR TITLE
helm-chart 0.4

### PR DESCRIPTION
**Pull request by yudong2015**
helm-chart 0.4

add am/im/notification services

### CODE REVIEW

File: openpitrix/templates/db/mysql.yaml [hunk 1]
```diff
@@ -13,8 +13,8 @@ spec:
   ports:
   - name: tcp
     protocol: TCP
-    port: 3306
-    targetPort: 3306
+    port: {{ .Values.mysql.service_port }}
```
Comment (chilianyi) [inline openpitrix/templates/db/mysql.yaml:16]: name: openpitrix-db
should be
{{ .Values.mysql.host }}

File: openpitrix/templates/etcd/etcd.yaml [hunk 2]
```diff
@@ -13,8 +13,8 @@ spec:
   ports:
   - name: tcp
     protocol: TCP
-    port: 2379
-    targetPort: 2379
+    port: {{ .Values.etcd.service_port }}
```
Comment (chilianyi) [inline openpitrix/templates/etcd/etcd.yaml:16]: name: openpitrix-etcd
should be {{ .Values.etcd.host }}

File: openpitrix/templates/openpitrix-am.yaml [hunk 3]
```diff
@@ -1,68 +1,66 @@
...
-        - name: OPENPITRIX_MYSQL_DATABASE
-          value: "iam"
-        - name: OPENPITRIX_MYSQL_PASSWORD
+        - name: AM_DB_HOST
+          value: {{ .Values.mysql.host | quote }}
+        - name: AM_DB_DATABASE
+          value: "am"
+        - name: AM_DB_LOG_MODE_ENABLE
+          value: {{ .Values.am.db_log_mode_enable | quote }}
+        - name: AM_DB_PASSWORD
```
Comment (chilianyi) [inline openpitrix/templates/openpitrix-am.yaml:69]: AM_IM_HOST, AM_LOG_LEVEL are needed.
OPENPITRIX_IAM_SECRET_KEY, IAM_INIT_CLIENT_ID,IAM_INIT_CLIENT_SECRET, IAM_INIT_ACCOUNT_EMAIL,  IAM_INIT_ACCOUNT_PASSWORD are useless, remove them.

File: openpitrix/templates/openpitrix-market.yaml [hunk 4]
```diff
@@ -41,11 +41,11 @@ spec:
       - name: wait-mysql
         image: busybox:1.28.4
         imagePullPolicy: IfNotPresent
-        command: ['sh', '-c', 'until nc -z openpitrix-db.{{ .Release.Namespace }}.svc 3306; do echo "waiting for mysql"; sleep 2; done;']
+        command: ['sh', '-c', 'until nc -z {{ .Values.mysql.host }}.{{ .Release.Namespace }}.svc {{ .Values.mysql.service_port }}; do echo "waiting for mysql"; sleep 2; done;']
```
Comment (chilianyi) [inline openpitrix/templates/openpitrix-market.yaml]: Delete market, no need now.

File: openpitrix/templates/openpitrix-notification.yaml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+          value: {{ .Values.grpc_show_error_cause | quote }}
+        - name: NOTIFICATION_LOG_LEVEL
+          value: {{ .Values.openpitrix_log_level }}
+        - name: NOTIFICATION_ETCD_ENDPOINTS
+          value: {{ .Values.etcd.endpoints | quote }}
+        - name: NOTIFICATION_MYSQL_HOST
+          value: {{ .Values.mysql.host | quote }}
+        - name: NOTIFICATION_MYSQL_DATABASE
+          value: "notification"
+        - name: MYSQL_ROOT_PASSWORD
```
Comment (chilianyi) [inline openpitrix/templates/openpitrix-notification.yaml]: Should be NOTIFICATION_MYSQL_PASSWORD

File: openpitrix/values.yaml [hunk 6]
```diff
@@ -30,7 +28,17 @@ memory_requests: 100
 cpu_limits: 500
 memory_limits: 500
 
-mysql_password: password
+etcd:
+  host: openpitrix-etcd
+  endpoints: openpitrix-etcd:2379
+  service_port: 2379
+  backend_port: 2379
```
Comment (chilianyi) [inline openpitrix/values.yaml:34]: endpoints is {{host}}:{{service_port}}, may be no need to define here.

File: openpitrix/templates/openpitrix-account.yaml [hunk 7]
```diff
@@ -1,60 +1,60 @@
...
+      - name: openpitrix-account-service
         image: {{ .Values.image }}
         imagePullPolicy: {{ .Values.image_pull_policy }}
         command:
-          - iam-service
+          - account-service
         ports:
         - containerPort: 9115
-          name: iam-service
+          name: account-service
```
Comment (chilianyi) [inline openpitrix/templates/openpitrix-account.yaml:57]: Add IAM_INIT_ACCOUNT_EMAIL to .Values

File: openpitrix/values.yaml [hunk 8]
```diff
@@ -47,6 +55,32 @@ global_config:
   max_repo_events: 20
 
 
+notification:
+  version: latest
+  image: openpitrix/notification:latest
+  flyway_image: openpitrix/notification:flyway
+  service_port: 9200
+  backend_port: 9200
```
Comment (chilianyi) [inline openpitrix/values.yaml]: notification port is 9201 in openpitrix

```
func NewClient() (*Client, error) {
	conn, err := manager.NewClient(constants.NotificationHost, constants.NotificationPort)
	if err != nil {
		return nil, err
	}
	return &Client{
		NotificationClient: nfpb.NewNotificationClient(conn),
	}, nil
}
```
backend_port should not be changed by user.

File: openpitrix/values.yaml [hunk 9]
```diff
@@ -47,6 +55,32 @@ global_config:
...
+
+
+im:
+  version: latest
+  image: kubespheredev/im:latest
+  flyway_image: kubespheredev/im:flyway
+  db_log_mode_enable: true
+  service_port: 9119
+  backend_port: 9119
```
Comment (chilianyi) [inline openpitrix/values.yaml]: backend_port should not be changed by user.

File: openpitrix/values.yaml [hunk 10]
```diff
@@ -47,6 +55,32 @@ global_config:
...
+
+
+am:
+  version: latest
+  image: openpitrix/iam:latest
+  flyway_image: openpitrix/iam:flyway
+  db_log_mode_enable: true
+  service_port: 9120
+  backend_port: 9120
```
Comment (chilianyi) [inline openpitrix/values.yaml]: backend_port should not be changed by user.

File: openpitrix/values.yaml [hunk 11]
```diff
@@ -30,7 +28,17 @@ memory_requests: 100
 cpu_limits: 500
 memory_limits: 500
 
-mysql_password: password
+etcd:
```
Comment (chilianyi) [inline openpitrix/values.yaml:31]: There are some new env in dashboard, have a look at docker-compose.yaml in openpitrix.

File: scripts/install-handlebars-renderer.sh [hunk 12]
```diff
@@ -2,7 +2,6 @@
 
 set -ev
 
-cd /opt
-git clone https://github.com/Hevienz/handlebars-renderer.git
-cd handlebars-renderer
-cargo install --path .
+# install handlerbars impl by python named pybars3
+# git@github.com:wbond/pybars3.git
+sudo pip install pybars3
```
Comment (chilianyi) [inline scripts/install-handlebars-renderer.sh]: Need pip? then should add to doc.

File: scripts/render-and-deploy-helm-chart.sh [hunk 13]
```diff
@@ -10,8 +10,8 @@ fi
 
 sed "s,VERSION,${VERSION},g" ./templates/values.json.tmpl > ./templates/values.json
 
-handlebars-renderer -i ./templates/values.yaml.handlebars -v ./templates/values.json -o ./openpitrix/values.yaml
-handlebars-renderer -i ./templates/Chart.yaml.handlebars -v ./templates/values.json -o ./openpitrix/Chart.yaml
+python ./scripts/handlebars-renderer.py -i ./templates/values.yaml.handlebars -v ./templates/values.json -o ./openpitrix/values.yaml
```
Comment (chilianyi) [inline scripts/render-and-deploy-helm-chart.sh:14]: Need python?

File: scripts/install-handlebars-renderer.sh (hunk 12 above)
Comment (yudong2015) [reply scripts/install-handlebars-renderer.sh]: Yes. Need to render handlebars template files with pybar3.

File: scripts/render-and-deploy-helm-chart.sh (hunk 13 above)
Comment (yudong2015) [reply scripts/render-and-deploy-helm-chart.sh:14]: Yes. Need to render handlebars template files with pybar3.

**Review by chilianyi — APPROVED** [review]

**Pull request merged**